tests/unit/test_utilities.py

```diff
@@ -358,3 +361,53 @@ def test_make_array_two_dimensional(self):
 
         self.assertEqual(len(output_array.shape), 2)
         np.testing.assert_array_equal(output_array, expected_output)
+
+
+class TestTransposeIfXdimLessThanYdim(TestCase):
```

Comment:
Thank you for adding a comprehensive set of tests for this new function! This is super helpful!

Reply:
Credit to @flamingbear on the tests as he wrote these before handing this ticket off to me.